// Copyright AGNTCY Contributors (https://github.com/agntcy)
// SPDX-License-Identifier: Apache-2.0

syntax = "proto3";

package core.v1alpha1;

// A specific skills that an agent is capable of performing.
// Specs: https://schema.oasf.agntcy.org/skills.
//
// Example (https://schema.oasf.agntcy.org/skills/contextual_comprehension)
message Skill {
  // Additional metadata for this skill.
  map<string, string> annotations = 1;
  // UID of the category.
  uint64 category_uid = 2;
  // UID of the class.
  uint64 class_uid = 3;
  // Optional human-readable name of the category.
  optional string category_name = 4;
  // Optional human-readable name of the class.
  optional string class_name = 5;
}
